app/controllers/api_export_controller.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+# ApiExportController: for exporting subjects via the API
+class ApiExportController < ActionController::API
+  before_action do
+    doorkeeper_authorize!(
+      :'system/Patient.read',
+      :'system/Patient.*'
+    )
+  end
+  before_action :set_client_app
+
+  def set_client_app
+    @current_client_app = OauthApplication.find_by(id: doorkeeper_token&.application_id)
+    head :unauthorized if @current_client_app.nil?
+  end
+
+  # Multi patient PHDC export
+  def nbs_patients
+    head(:not_acceptable) && return unless request.headers['Accept']&.include?('application/zip')
+
+    search_params = params.slice('workflow', 'monitoring', 'caseStatus', 'updatedAt')
```

Comment:
Mostly done in https://github.com/SaraAlert/SaraAlert/pull/823/commits/ead02e3fc1bcc79b2c717085f526b9b2cf59c480. I didn't use `params.slice('workflow', 'monitoring', 'caseStatus', 'updatedAt')` directly, since I still want to be able to check if the params are empty to set the initial value for `patients`. This is done so that an empty search returns no patients, but I could be convinced that an empty search should return all of the accessible patients